Bottom: 97326df669c5fd18a6d9f79776adefea5dbfb809
Top:    69a844b89d8aa5c51d33d674e587ec4703d1b0cc
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 15:55:54 +0000

polygon.c: Refactor NoHolesPolygonDicer to call poly_Copy0 rather than poly_Copy1

poly_Copy1 is much more of an internal function to polygon1.c, and this is its
only external caller. Replace this occurance with poly_Copy0, which is intended
to be the public API.


---

diff --git a/src/polygon.c b/src/polygon.c
index cbe398e..b94fdb9 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1823,9 +1823,8 @@ NoHolesPolygonDicer (PolygonType *p, const BoxType * clip,
 {
   POLYAREA *main_contour, *cur, *next;
 
-  main_contour = poly_Create ();
   /* copy the main poly only */
-  poly_Copy1 (main_contour, p->Clipped);
+  poly_Copy0 (&main_contour, p->Clipped);
   /* clip to the bounding box */
   if (clip)
     {
